### PR TITLE
Issue 243: Convert pekko-samples `cluster-client` java/scala to pekko

### PR DESCRIPTION
**Pull request by sam-byng**
incubator-pekko issue : https://github.com/apache/incubator-pekko/issues/243

**Method:**
Steps taken from updated migration guide in PR:
In your `pom.xml` and `build.sbt` files:
* Update dependency Akka version to the latest Pekko version.
* for Pekko jar dependencies, the groupId is `org.apache.pekko` instead of `com.typesafe.akka` or `com.lightbend.akka.<package_name>`
* the jar names (i.e. the `artifactId` field in `pom.xml`) start with "pekko" instead of "akka" - e.g. pekko-actor_2.13.jar instead of akka-actor_2.13.jar
* Alpakka equivalent is "pekko-connectors" - e.g. pekko-connectors-kafka_2.13.jar instead of alpakka-kafka_2.13.jar

In source code files:
* Pekko packages start with `org.apache.pekko` instead of `akka` - e.g. `import org.apache.pekko.actor` instead of `import akka.actor`
* Where class names have "Akka" in the name, the Pekko ones have "Pekko" - e.g. `PekkoException` instead of `AkkaException`
* Config changes use `pekko` prefix instead of `akka`

In config files for your module:
* Configs use `pekko` prefix instead of `akka`

In Documentation:
* Where appropriate, text should refer to `Pekko` instead of `Akka`
* URLs linking to files in https://doc.akka.io/docs/akka/ should instead point at https://pekko.apache.org/docs/pekko/

*Note: I checked the lightbend.akka.grpc project and those packages are going to org.apache.pekko so assuming this is the case for all

### CODE REVIEW

**Comment by pjfanning** [comment]
Thanks. The CI build failed unfortunately. If you look in the in the .GitHub directory, the CI builds are controlled by yml files. You'll need to update the dir names to match your changes.

**Comment by sam-byng** [comment]
Outstanding questions for review:

- I've updated all LICENSE files to contain the apache license - correct?
- I've updated all doc links from `doc.akka.io/docs/akka/2.6/` to `pekko.apache.org/docs/pekko/current/` - correct?
- I've not changed doc links for developer.lightbend.com/docs/ because i haven't found equivalent yet

**Comment by pjfanning** [comment]
Leave the licenses as they were. I think we won't change them at all. I need to discuss with the ASF Incubator.

**Comment by pjfanning** [comment]
I'll review the other changes later. I only have a mobile at the moment.

**Comment by sam-byng** [comment]
> Thanks. The CI build failed unfortunately. If you look in the in the .github directory, the CI builds are controlled by yml files. You'll need to update the dir names to match your changes.

Done

**Comment by sam-byng** [comment]
https://github.com/apache/incubator-pekko/pull/249 I've updated migration docs in line with this

**Comment by pjfanning** [comment]
Big -1 on this PR. Will block its progress until the CI failures are addressed in this PR.

**Review by pjfanning — CHANGES_REQUESTED** [review]
Blocking until I can review both PRs. Am 100% befuddled on what reason exists to have 2 overlapping PRs.

**Comment by pjfanning** [comment]
Let's concentrate on https://github.com/apache/incubator-pekko-samples/pull/6 first - I have comments on that and when that is merged, similar changes can be applied here.

**Comment by pjfanning** [comment]
CI is broken - it appears that you are missing the required repository definition to get the pekko snapshot jars from repository.apache.org.

In sbt, this looks like:
```
resolvers += "Apache Nexus Snapshots".at("https://repository.apache.org/content/repositories/snapshots/")
```

File: pekko-sample-cluster-scala/src/main/resources/application.conf [hunk 1]
```diff
@@ -14,9 +14,9 @@ akka {
   }
   cluster {
     seed-nodes = [
-      "akka://ClusterSystem@127.0.0.1:25251",
-      "akka://ClusterSystem@127.0.0.1:25252"]
-    downing-provider-class = "akka.cluster.sbr.SplitBrainResolverProvider"
+      "pekko://ClusterSystem@127.0.0.1:25251",
```
Comment (pjfanning) [inline pekko-sample-cluster-scala/src/main/resources/application.conf:17]: could you use 17356 and 17357? Pekko Artery port default is 17355

File: pekko-sample-cluster-client-grpc-scala/project/plugins.sbt [hunk 2]
```diff
@@ -1,3 +1,3 @@
-addSbtPlugin("com.lightbend.akka.grpc" % "sbt-akka-grpc" % "0.8.4")
+addSbtPlugin("org.apache.pekko" % "sbt-pekko-grpc" % "0.8.4")
```
Comment (sam-byng) [inline pekko-sample-cluster-client-grpc-scala/project/plugins.sbt:1]: Are we not publishing the sbt-pekko-grpc plugins yet?

I thought we were, looking at the successful pipeline : https://github.com/apache/incubator-pekko-grpc/actions/runs/4425762836/jobs/7761223926 . However, even with updated grpc version to `0.0.0-15-3d8bff9d-SNAPSHOT`, sbt multi-jvm:test still failed due to the package not being found. Therefore i've reverted back to the akka dependency.

Error message:

```
[error] (update) sbt.librarymanagement.ResolveException: Error downloading org.apache.pekko:sbt-pekko-grpc_2.12_1.0;sbtVersion=1.0;scalaVersion=2.12:0.0.0-15-3d8bff9d-SNAPSHOT
[error]   Not found
[error]   Not found
[error]   not found: https://repo1.maven.org/maven2/org/apache/pekko/sbt-pekko-grpc_2.12_1.0_2.12_1.0/0.0.0-15-3d8bff9d-SNAPSHOT/sbt-pekko-grpc_2.12_1.0-0.0.0-15-3d8bff9d-SNAPSHOT.pom
[error]   not found: /home/developer/.ivy2/localorg.apache.pekko/sbt-pekko-grpc_2.12_1.0/scala_2.12/sbt_1.0/0.0.0-15-3d8bff9d-SNAPSHOT/ivys/ivy.xml
[error]   not found: https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.apache.pekko/sbt-pekko-grpc_2.12_1.0/scala_2.12/sbt_1.0/0.0.0-15-3d8bff9d-SNAPSHOT/ivys/ivy.xml
[error]   not found: https://repo.typesafe.com/typesafe/ivy-releases/org.apache.pekko/sbt-pekko-grpc_2.12_1.0/scala_2.12/sbt_1.0/0.0.0-15-3d8bff9d-SNAPSHOT/ivys/ivy.xml
```

Comment (sam-byng) [reply pekko-sample-cluster-client-grpc-scala/project/plugins.sbt:1]: I can see the referenced files in https://repository.apache.org/content/repositories/snapshots/org/apache/pekko/sbt-pekko-grpc_2.12_1.0/0.0.0-15-3d8bff9d-SNAPSHOT/ but this is not being pulled

Comment (pjfanning) [reply pekko-sample-cluster-client-grpc-scala/project/plugins.sbt:1]: you seem not to have defined the resolver for repository.apache.org unless you clipped the error messages above - 4 repos searched but not the right one

**Comment by sam-byng** [comment]
Status: The `pekko-sample-cluster-client-grpc-` packages are failing to compile due to issues with ClusterClient objects in pekko.

- Running sbt multi-jvm:test
developer@DESKTOP-ISCIMSK:~/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-scala$ sbt multi-jvm:test
[info] welcome to sbt 1.5.6 (Ubuntu Java 11.0.18)
[info] loading settings for project pekko-sample-cluster-client-grpc-scala-build from plugins.sbt ...
[info] loading project definition from /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-scala/project
[info] loading settings for project pekko-sample-cluster-client-grpc-scala from build.sbt ...
[info] set current project to pekko-sample-cluster-client-grpc-scala (in build file:/home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-scala/)
[warn] sbt 0.13 shell syntax is deprecated; use slash syntax instead: Multi-jvm / test
[info] compiling 15 Scala sources to /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-scala/target/scala-2.13/classes ...
[error] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-scala/src/main/scala/sample/cluster/client/grpc/ClusterClient.scala:54:43: could not find implicit value for parameter sys: org.apache.pekko.actor.ClassicActorSystemProvider
[error]     ClusterClientReceptionistServiceClient(settings.grpcClientSettings)
[error]                                           ^
[warn] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-scala/src/main/scala/sample/cluster/client/grpc/ClusterClient.scala:53:18: local val ec in method createClientStub is never used
[warn]     implicit val ec: ExecutionContext = mat.executionContext
[warn]                  ^
[warn] one warning found
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
[error] Total time: 10 s, completed Mar 15, 2023, 7:59:21 PM

- On the java repo, running mvn test also gives failures of a similar kind:
- ```
$ mvn test
...
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientSerialization.java:[19,2] error: cannot find symbol
  symbol:   class Payload
  location: class ClusterClientSerialization
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientSerialization.java:[31,28] error: cannot find symbol
  symbol:   class Payload
  location: class ClusterClientSerialization
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[17,51] error: cannot find symbol
  symbol: class ClusterClientReceptionistService
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[33,48] error: cannot find symbol
  symbol:   class Req
  location: class ClusterClientReceptionistGrpcImpl
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[33,16] error: cannot find symbol
  symbol:   class Rsp
  location: class ClusterClientReceptionistGrpcImpl
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[86,38] error: cannot find symbol
  symbol:   class SendReq
  location: class ClusterClientReceptionistGrpcImpl
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[86,25] error: cannot find symbol
  symbol:   class Rsp
  location: class ClusterClientReceptionistGrpcImpl
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[134,17] error: cannot find symbol
  symbol:   class ClusterClientReceptionistServiceClient
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[141,6] error: cannot find symbol
  symbol:   class ClusterClientReceptionistServiceClient
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[216,4] error: cannot find symbol
  symbol:   class ClusterClientReceptionistServiceClient
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[233,16] error: cannot find symbol
  symbol:   class ClusterClientReceptionistServiceClient
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientSerialization.java:[24,11] error: cannot find symbol
  symbol:   variable Payload
  location: class ClusterClientSerialization
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[32,2] error: method does not override or implement a method from a supertype
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[45,8] error: cannot find symbol
  symbol:   class Payload
  location: class ClusterClientReceptionistGrpcImpl
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[46,15] error: cannot find symbol
  symbol:   variable Rsp
  location: class ClusterClientReceptionistGrpcImpl
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[50,37] error: package Req does not exist
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[51,12] error: cannot find symbol
  symbol:   class SendReq
  location: class ClusterClientReceptionistGrpcImpl
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[57,44] error: package Req does not exist
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[58,12] error: cannot find symbol
  symbol:   class SendToAllReq
  location: class ClusterClientReceptionistGrpcImpl
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[63,44] error: package Req does not exist
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[64,12] error: cannot find symbol
  symbol:   class PublishReq
  location: class ClusterClientReceptionistGrpcImpl
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[85,2] error: method does not override or implement a method from a supertype
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[94,8] error: cannot find symbol
  symbol:   class Payload
  location: class ClusterClientReceptionistGrpcImpl
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[95,15] error: cannot find symbol
  symbol:   variable Rsp
  location: class ClusterClientReceptionistGrpcImpl
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionistGrpcImpl.java:[99,24] error: cannot find symbol
  symbol:   class Rsp
  location: class ClusterClientReceptionistGrpcImpl
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionist.java:[66,6] error: cannot find symbol
  symbol:   variable ClusterClientReceptionistServiceHandlerFactory
  location: class ClusterClientReceptionist
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[136,11] error: cannot find symbol
  symbol:   variable ClusterClientReceptionistServiceClient
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[166,14] error: cannot find symbol
  symbol:   class Payload
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[167,21] error: cannot find symbol
  symbol:   variable Req
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[168,25] error: cannot find symbol
  symbol:   variable SendReq
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[175,14] error: cannot find symbol
  symbol:   class Payload
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[176,21] error: cannot find symbol
  symbol:   variable Req
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[177,30] error: cannot find symbol
  symbol:   variable SendToAllReq
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[183,14] error: cannot find symbol
  symbol:   class Payload
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[184,21] error: cannot find symbol
  symbol:   variable Req
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[185,28] error: cannot find symbol
  symbol:   variable PublishReq
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[219,4] error: cannot find symbol
  symbol:   class Payload
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[220,4] error: cannot find symbol
  symbol:   class SendReq
  location: class ClusterClient
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[220,22] error: cannot find symbol
  symbol:   variable SendReq
  location: class ClusterClient
```

**Comment by pjfanning** [comment]
This PR is gigantic. Can you take the stuff that works and put in a new PR (or PRs)? We can then look at the remaining broken stuff.

**Comment by sam-byng** [comment]
However, now both compilations are failing with similar errors: (same for JDK8 and JDK11)

java:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project
 pekko-sample-cluster-client-grpc-java: Compilation failure: Compilation failure:
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClientReceptionist.java:[66,59] error: incompatible types: org.apache.pekko.japi.function.Function<HttpRequest,CompletionStage<HttpResponse>> cannot be converted to org.apache.pekko.japi.Function<HttpRequest,CompletionStage<HttpResponse>>
[ERROR] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-java/src/main/java/sample/cluster/client/grpc/ClusterClient.java:[136,49] error: no suitable method found for create(GrpcClientSettings,Materializer,ExecutionContextExecutor)
[ERROR]     method ClusterClientReceptionistServiceClient.create(GrpcClientSettings,ClassicActorSystemProvider) is not applicable
[ERROR]       (actual and formal argument lists differ in length)
[ERROR]     method ClusterClientReceptionistServiceClient.create(GrpcChannel,ClassicActorSystemProvider) is not applicable
[ERROR]       (actual and formal argument lists differ in length)
```

scala:
```
[info]   Compilation completed in 10.74s.
[error] /home/developer/documents/projects/mvm/pekko/sb5-fork/incubator-pekko-samples/pekko-sample-cluster-client-grpc-scala/src/main/scala/sample/cluster/client/grpc/ClusterClient.scala:54:43: could not find implicit value for parameter sys: org.apache.pekko.actor.ClassicActorSystemProvider
[error]     ClusterClientReceptionistServiceClient(settings.grpcClientSettings)
[error]
```

**Comment by pjfanning** [comment]
So for Scala, you seem to need to declare an extra implicit val. As per the compile error message. We can look at pekko-grpc later to see why this has become the case

File: akka-sample-cluster-client-grpc-java/pom.xml [hunk 3]
```diff
@@ -1,99 +0,0 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0"
-  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0
-                      http://maven.apache.org/xsd/maven-4.0.0.xsd">
-  <modelVersion>4.0.0</modelVersion>
-
-  <properties>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <akka.version>2.6.20</akka.version>
-    <akka.grpc.version>0.8.4</akka.grpc.version>
```
Comment (sam-byng) [inline akka-sample-cluster-client-grpc-java/pom.xml:10]: I think that the issue here is that this samples repository was using Akka GRPC at version 0.8.4 (april 2020). Akka GRPC is now at 2.3.0
https://github.com/akka/akka-grpc/tags?after=v1.1.0

Our pekko-grpc is forked from akka-grpc last committed to in september 2022 -- Akka gRPC 2.1.6 --https://github.com/apache/incubator-pekko-grpc/commits/forked-from-akka 
https://github.com/akka/akka-grpc/tree/v2.1.6

So obviously between these two versions there have been breaking changes in the way that ServiceClients (?) are handled, which we have got to integrate into the samples library now.

**Comment by sam-byng** [comment]
See #16

**Comment by pjfanning** [comment]
I'd prefer to at least get the sample migrated to prove it can be done - we can decide later if we want to prune less relevant samples.

**Review by pjfanning — APPROVED** [review]
looks ok to merge

**Pull request merged**